src/layouts/UserLayout.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {Navigate } from 'react-router-dom';
+function UserLayout(){
+  const token: string | null = localStorage.getItem('user_profile_token');
+  
+  if (!token) {
+    return <Navigate to="/auth" replace />;
```

Comment:
token will be needed to be saved in the redux store as it will be used in multiple components

Reply:
yes, of course. I just used it here initially and will move it to the redux store.